pravega-client-examples/src/main/java/io/pravega/example/secure/SecureReader.java

```diff
@@ -0,0 +1,186 @@
+         *      deployment, it is recommended to keep this on.
+         *
+         * Note about setting the client config for auth:
+         *    - The client config below is configured with an object of DefaultCredentials class. The user name
+         *      and password arguments passed to the object represent the credentials used for authentication
+         *      and authorization. The assumption we are making here is that the username is valid on the server,
+         *      the password is correct and the username has all the permissions necessary for performing the
+         *      subsequent operations.
+         */
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(this.controllerURI) // "tls://localhost:9090"
+
+                // TLS-related client-side configuration
+                .trustStore(this.truststorePath)
+                .validateHostName(this.validateHostName)
+
+                // Auth-related client-side configuration
+                .credentials(new DefaultCredentials(this.password, this.username))
+                .build();
+
+        System.out.println("Done creating a client config.");
+
+        // Everything below depicts the usual flow of reading events. All client-side security configuration is
+        // done through the ClientConfig object as shown above.
+
+        ClientFactory clientFactory = null;
+        ReaderGroupManager readerGroupManager = null;
+        EventStreamReader<String> reader = null;
+        try {
+            ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                    .stream(Stream.of(this.scope, this.stream))
+                    .disableAutomaticCheckpoints()
+                    .build();
+            System.out.println("Done creating a reader group config with specified scope: [" +
+                    this.scope +"] and stream name: [" + this.stream + "].");
+
+            String readerGroupName = UUID.randomUUID().toString().replace("-", "");
+            readerGroupManager = ReaderGroupManager.withScope(this.scope, clientConfig);
+            readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig);
+            System.out.println("Done creating a reader group with specified name  and config.");
+
+            clientFactory = ClientFactory.withScope(this.scope, clientConfig);
+            System.out.println("Done creating a client factory with the specified scope and client config.");
+
+            reader = clientFactory.createReader("readerId", readerGroupName,
+                    new JavaSerializer<>(), ReaderConfig.builder().build());
+            System.out.println("Done creating a reader.");
+
+            String readMessage = reader.readNextEvent(2000).getEvent();
+            System.out.println("Done reading an event: [" + readMessage + "].");
+
+        } finally {
```

Comment:
Please consider using try-with-resources option to avoid these calls.

Reply:
I intentionally avoided using try-with-resources to close resources automatically here, in order to:
* Avoid deep nesting and to ensure that most of the code reads left-justified. This is to make the code more readable. 
* Order of resource closure explicit and as simple as possible to understand. 

All in all, since the code is for consumption of client application developers, I've favored readability and understandability over brevity here. 

@RaulGracia and I had the same discussion over here: https://github.com/pravega/pravega-samples/pull/182#discussion_r262161189.